File: Python/visitCardManagement.py
visit_cards = []

def manage_visit_cards():
    global visit_cards
    
    while True:
        sub_menu_visit_cards = int(input("Option de gestion des cartes de visite (1. Ajouter, 2. Consulter, 3. Supprimer, 4. Retour) : "))

        if sub_menu_visit_cards not in [1, 2, 3, 4]:
            print("Veuillez saisir une option valable")
            continue

        if sub_menu_visit_cards == 1:
            # Ajout d'une carte de visite
            nom = input("Nom sur la carte de visite : ")
            prenom = input("Prénom sur la carte de visite : ")
            entreprise = input("Entreprise sur la carte de visite : ")
            telephone = input("Numéro de téléphone sur la carte de visite : ")

            if nom.strip() == "" or prenom.strip() == "" or entreprise.strip() == "" or telephone.strip() == "":
                print("Veuillez remplir tous les champs de la carte de visite.")
            else:
                new_visit_card = {
                    "nom": nom,
                    "prenom": prenom,
                    "entreprise": entreprise,
                    "telephone": telephone,
                }

                visit_cards.append(new_visit_card)
                print("Carte de visite ajoutée avec succès.")
        elif sub_menu_visit_cards == 2:
            # Consultation des cartes de visite
            if visit_cards:
                print("Cartes de visite enregistrées :")
                for card in visit_cards:
                    print(f"Nom : {card['nom']}, Prénom : {card['prenom']}, Entreprise : {card['entreprise']}, Téléphone : {card['telephone']}")
            else:
                print("Aucune carte de visite enregistrée.")
        elif sub_menu_visit_cards == 3:
            # Suppression d'une carte de visite
            if visit_cards:
                for card in visit_cards:
                    print(f"Nom : {card['nom']}, Prénom : {card['prenom']}, Entreprise : {card['entreprise']}, Téléphone : {card['telephone']}")

                card_to_delete = input("Nom de la carte de visite à supprimer (Appuyez sur Entrée pour annuler) : ")
                if card_to_delete.strip() == "":
                    print("Suppression annulée.")
                    continue
                index = next((i for i, card in enumerate(visit_cards) if card["nom"] == card_to_delete), -1)

                if index != -1:
                    visit_cards.pop(index)
                    print("Carte de visite supprimée avec succès.")
                else:
                    print("Carte de visite non trouvée. Veuillez vérifier le nom de la carte.")
            else:
                print("Aucune carte de visite enregistrée.")
        elif sub_menu_visit_cards == 4:
            print("Retour au menu précédent.")
            break

if __name__ == "__main__":
    manage_visit_cards()

File: JS/visitCardManagement.js
const prompt = require("prompt-sync")();
let visitCards = [];

function manageVisitCards() {

    while (true) {
        let subMenuVisitCards = +prompt("Option de gestion des cartes de visite (1. Ajouter, 2. Consulter, 3. Supprimer, 4. Retour) : ");

        if (![1, 2, 3, 4].includes(subMenuVisitCards)) {
            console.log("Veuillez saisir une option valable");
            continue;
        }

        if (subMenuVisitCards === 1) {
            // Ajout d'une carte de visite
            let nom = prompt("Nom sur la carte de visite : ");
            let prenom = prompt("Prénom sur la carte de visite : ");
            let entreprise = prompt("Entreprise sur la carte de visite : ");
            let telephone = prompt("Numéro de téléphone sur la carte de visite : ");

            if (nom.trim() === "" || prenom.trim() === "" || entreprise.trim() === "" || telephone.trim() === "") {
                console.log("Veuillez remplir tous les champs de la carte de visite.");
            } else {
                let newVisitCard = {
                    nom: nom,
                    prenom: prenom,
                    entreprise: entreprise,
                    telephone: telephone,
                };

                visitCards.push(newVisitCard);
                console.log("Carte de visite ajoutée avec succès.");
            }
        } else if (subMenuVisitCards === 2) {
            // Consultation des cartes de visite
            if (visitCards.length > 0) {
                console.log("Cartes de visite enregistrées :");
                for (let card of visitCards) {
                    console.log(`Nom : ${card.nom}, Prénom : ${card.prenom}, Entreprise : ${card.entreprise}, Téléphone : ${card.telephone}`);
                }
            } else {
                console.log("Aucune carte de visite enregistrée.");
            }
        } else if (subMenuVisitCards === 3) {
            // Suppression d'une carte de visite
            if (visitCards.length > 0) {
                for (let card of visitCards) {
                    console.log(`Nom : ${card.nom}, Prénom : ${card.prenom}, Entreprise : ${card.entreprise}, Téléphone : ${card.telephone}`);
                }

                let cardToDelete = prompt("Nom de la carte de visite à supprimer (Appuyez sur Entrée pour annuler) : ");
                if (cardToDelete.trim() === "") {
                    console.log("Suppression annulée.");
                    continue;
                }
                let index = visitCards.findIndex(card => card.nom === cardToDelete);

                if (index !== -1) {
                    visitCards.splice(index, 1);
                    console.log("Carte de visite supprimée avec succès.");
                } else {
                    console.log("Carte de visite non trouvée. Veuillez vérifier le nom de la carte.");
                }
            } else {
                console.log("Aucune carte de visite enregistrée.");
            }
        } else if (subMenuVisitCards === 4) {
            console.log("Retour au menu précédent.");
            break;
        }
    }
}

module.exports = {
    manageVisitCards
};